packages/analytics-core/src/plugins/destination.ts

```diff
@@ -241,11 +251,14 @@ export class Destination implements DestinationPlugin {
       // log intermediate event status before retry
       this.config.loggerProvider.warn(getResponseBodyString(res));
     }
+
+    void this.updateEvent(dropEventSet);
```

Comment:
CASE 3: when events have an invalid response.

Reply:
When invalid response, we drop events 
- [When body has missing field or invalid api key](https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/analytics-core/src/plugins/destination.ts#L219-L222)
- [When events have invalid or missing fields](https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/analytics-core/src/plugins/destination.ts#L233-L236) . 
Both cases will return early and not execute `updateEvent()` here.

If code is executed here, events should be retried.